packages/compass-components/src/components/workspace-tabs/tab.tsx

```diff
@@ -185,25 +190,34 @@ const workspaceTabTooltipStyles = css({
   textWrap: 'wrap',
 });
 
-type TabProps = {
+// The plugins provide these essential props use to render the tab.
+// The workspace-tabs component provides the other parts of TabProps.
+export type WorkspaceTabPluginProps = {
   connectionName?: string;
   type: string;
-  title: string;
+  title: React.ReactNode;
+  isNonExistent?: boolean;
+  iconGlyph: GlyphName | 'Logo' | 'Server';
+  tooltip?: [string, string][];
+  tabTheme?: Partial<TabTheme>;
```

Comment:
We could get rid of this prop and instead use a provider.
Here's a branch with that provider: https://github.com/mongodb-js/compass/compare/main...tab-theme-context-provider-example 
I didn't include it in these changes, as it is something we could do as a follow up.

Reply:
Makes sense to me, and then we can also add it to all the shared workspace providers. Then we can also use these colors inside the workspaces UI easily if we need to